src/App.tsx

```diff
@@ -1,26 +1,236 @@
+        for (const address of this.state.highlights) {
+            if (!this.isPermanent(address)) {
+                newValues[address] = null;
+                /**
+                  * //TODO: if marking exist on first delete click
+                  * delete only value and show markings on second
+                  * click delete markings
+                  */
+            }
+        }
+        this.setState({
+            ...this.state,
+            values: newValues,
+        });
+    }
+
+    public render() {
+        return(
+            <div className="game">
+                <div className="board">{this.renderSquares()}</div>
+                <div className="button-box">
+                    <div className="button-box-top">
+                        <ControlButtons/>
+                        <Numpad
+                        onClickNum={(i: SquareValue) => {this.normalMarkSelectedSquares(i); }}
+                        onClickDel={() => {this.deleteSelectedSquares(); }}
+                        />
+                    </div>
+                    <div className="button-box-bot">
+                        <Button variant="contained" color="primary">
+                          R E S T A R T
+                        </Button>
+                        <Button variant="contained" color="secondary">
+                          C H E C K
+                        </Button>
+                    </div>
+                </div>
+            </div>
+        );
+    }
+
+    public renderSquares() {
+        const list = [];
+        for (let i = 0; i < 81; i++) {
+            list.push(this.renderSquare(i));
+        }
+        return list;
+    }
+
+    public renderSquare(i: SquareAddress) {
+        const isHighlighted = this.state.highlights.has(i);
+        const isContradicting = this.state.contradicts.has(i);
+        const marks = i % 9 + 1 == 7 ? [i % 1 as SquareValue, i % 5 + 1 as SquareValue, i % 3 + 1 as SquareValue, i % 8 + 1 as SquareValue] : [];
+        return <Square
+            value={this.isPermanent(i) ? this.props.permanentValues[i] : this.state.values[i]}
+            isPermanent={this.isPermanent(i)}
+            isHighlighted = {isHighlighted}
+            isContradicting={isContradicting}
+            markings={marks}
+            onClick={() => { this.toggleSelectSquare(i);}}
+        />;
+    }
+}
+
+interface SquareProps {
+    isPermanent: boolean;
+    isHighlighted: boolean;
+    isContradicting: boolean;
+    value: SquareValue | null;
+    markings: SquareValue[];
+    onClick: () => void;
+}
+
+type SquareValue = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 ;
+
+class Square extends React.Component<SquareProps, {}> {
+    public getMarks(): (SquareValue | null)[] {
+        const lst: (SquareValue | null)[] = [];
+        for (let i = 0; i < 9; i++) {
+            lst[i] = this.props.markings.includes((i+1) as SquareValue) ? ((i+1) as SquareValue) : null;
+        }
+        return lst;
+    }
+    public renderMarkings() {
```

Comment:
Collapse the functionality of getMarks and rederMarkings into a single function.